frontend/src/framework/internal/components/TopBar/topBar.tsx

```diff
@@ -55,3 +78,363 @@ function LogoWithText(): React.ReactNode {
+            <Typography variant="body_short" className="font-light">
+                (snapshot)
+            </Typography>
+            <Tooltip title="This session is a snapshot and cannot be edited.">
+                <Lock fontSize="inherit" />
+            </Tooltip>
+        </>
+    );
+}
+
+type SessionTitleProps = {
+    workbench: Workbench;
+};
+
+function SessionTitle(props: SessionTitleProps): React.ReactNode {
+    const activeSession = usePublishSubscribeTopicValue(
+        props.workbench.getSessionManager(),
+        WorkbenchSessionManagerTopic.ACTIVE_SESSION,
+    );
+    const metadata = usePublishSubscribeTopicValue(activeSession!, PrivateWorkbenchSessionTopic.METADATA);
+    const isPersisted = usePublishSubscribeTopicValue(activeSession!, PrivateWorkbenchSessionTopic.IS_PERSISTED);
+
+    const persistenceInfo = usePublishSubscribeTopicValue(
+        props.workbench.getSessionManager().getPersistenceOrchestrator()!,
+        PersistenceOrchestratorTopic.PERSISTENCE_INFO,
+    );
+
+    const hasChanges = (persistenceInfo.hasChanges && persistenceInfo.lastPersistedMs !== null) || !isPersisted;
+
+    return (
+        <>
+            <Category fontSize="inherit" className="mr-1" />
+            <Typography
+                variant="h5"
+                className={resolveClassNames("overflow-ellipsis min-w-0 whitespace-nowrap", {
+                    italic: !isPersisted,
+                })}
+            >
+                <Tooltip
+                    title={
+                        <div className="whitespace-normal text-base">
+                            <h3 className="text-lg">{metadata.title}</h3>
+                            {metadata.description && (
+                                <>
+                                    <hr className="h-px mb-2 bg-white/25" />
+                                    <p className="text-sm whitespace-pre-wrap">{metadata.description}</p>
+                                </>
+                            )}
+                        </div>
+                    }
+                    placement="bottom"
+                    enterDelay="medium"
+                >
+                    <span className="truncate">{metadata.title}</span>
+                </Tooltip>
+                <Tooltip title="You have unsaved changes">
+                    <span className="text-amber-600 ml-2 text-2xl">{hasChanges ? "*" : " "}</span>
+                </Tooltip>
+            </Typography>
+        </>
+    );
+}
+
+type SessionFromSnapshotButtonProps = {
+    workbench: Workbench;
+};
+
+function SessionFromSnapshotButton(props: SessionFromSnapshotButtonProps): React.ReactNode {
+    const handleClick = () => {
+        props.workbench.getSessionManager().convertSnapshotToSession();
+    };
+
+    return (
+        <div className="p-2 flex items-center text-sm gap-4">
+            <TopBarButton onClick={handleClick} title="Make a new session of the current snapshot" variant="contained">
```

Comment:
When I'm testing the snapshot seems to be created form the current state of session, i.e. not based on saved/last persisted state.

Thus we perhaps should add text that clarifies that snapshot is created from session with unsaved changes - to prevent confusion? When i tested my self, I forgot to save the session, thus my session and snapshot was not in synch and I got a bit confused.

Alternative: Clarify/notify the user about this in the dialog for creating snapshot?